### ISSUE TITLE
no function after updating from 0.11 to 0.20

### ISSUE THREAD

**Issue by jubie25** [p0l0/hagruenbeck_cloud#135]
## System Information

version | core-2024.10.0
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.12.4
os_name | Linux
os_version | 6.6.46-haos
arch | x86_64
timezone | Europe/Berlin
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
HACS Data | ok
GitHub API Calls Remaining | 5000
Installed Version | 2.0.1
Stage | running
Available Repositories | 1430
Downloaded Repositories | 10

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | false
-- | --
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 13.1
-- | --
update_channel | stable
supervisor_version | supervisor-2024.09.1
agent_version | 1.6.0
docker_version | 26.1.4
disk_total | 62.3 GB
disk_used | 31.4 GB
healthy | true
supported | true
host_connectivity | true
supervisor_connectivity | false
ntp_synchronized | true
virtualization | kvm
board | ova
supervisor_api | ok
version_api | ok
installed_addons | Studio Code Server (5.16.3), Terminal & SSH (9.15.0), UniFi Network Application (3.3.0), Mosquitto broker (6.4.1), File editor (5.8.0), MariaDB (2.7.1), phpMyAdmin (0.10.0), ESPHome (2024.9.2)

</details>

<details><summary>Dashboards</summary>

dashboards | 5
-- | --
resources | 7
views | 23
mode | storage

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | 23. September 2024 um 07:52
-- | --
current_recorder_run | 3. Oktober 2024 um 15:50
estimated_db_size | 2085.11 MiB
database_engine | mysql
database_version | 10.11.6

</details>

### Checklist

- [X] I have enabled debug logging for my installation.
- [X] I have filled out the issue template to the best of my ability.
- [X] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).
- [X] This issue is not a duplicate issue of any [previous issues](https://github.com/p0l0/hagruenbeck_cloud/issues?q=is%3Aissue+label%3A%22Bug%22+)..

### Describe the issue

After updateing from 0.11 to 0.20 the integration stopped working: Error in configuration.
1. searched for backups. But this update didn' create one.
2. deleted from HACS and reloaded 0.11 from HACS
3. Lead to error "Der Konfigurationsfluss konnte nicht geladen werden: 500 Internal Server Error Server got itself in trouble"
4. tried 0.20 again, same error
5. searched for configuration data to manually delete them - not found in file system 
Of course many reboots were done. It now ends with having no Grünbeck integration at all.

-> search for a way to totally reset all data for the Grünbeck Cloud integration
-> please find the bug that made the integration dysfunctional
-> maybe the integration can create a backup when being updated?

### Reproduction steps

1. see above
2.
3.
...

### Debug logs

```text
These logs are collected during my several tries to get the integration working again:

Logger: homeassistant.util.package
Quelle: util/package.py:150
Erstmals aufgetreten: 15:50:38 (3 Vorkommnisse)
Zuletzt protokolliert: 15:50:39

Unable to install package pygruenbeck_cloud==0.1.0: × No solution found when resolving dependencies: ╰─▶ Because pygruenbeck-cloud==0.1.0 depends on yarl>=1.9.4,<1.10.dev0 and yarl==1.13.1, we can conclude that pygruenbeck-cloud==0.1.0 cannot be used. And because you require pygruenbeck-cloud==0.1.0, we can conclude that your requirements are unsatisfiable.




Logger: aiohttp.server
Quelle: /usr/local/lib/python3.12/site-packages/aiohttp/web_protocol.py:448
Erstmals aufgetreten: 15:50:39 (1 Vorkommnisse)
Zuletzt protokolliert: 15:50:39

Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/aiohttp/web_protocol.py", line 477, in _handle_request
    resp = await request_handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiohttp/web_app.py", line 559, in _handle
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiohttp/web_middlewares.py", line 117, in impl
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/security_filter.py", line 92, in security_filter_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/forwarded.py", line 210, in forwarded_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/request_context.py", line 26, in request_context_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/ban.py", line 85, in ban_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/auth.py", line 242, in auth_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/headers.py", line 32, in headers_middleware
    response = await handler(request)
               ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/http.py", line 73, in handle
    result = await handler(request, **request.match_info)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/decorators.py", line 81, in with_admin
    return await func(self, request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/data_validator.py", line 74, in wrapper
    return await method(view, request, data, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/config/config_entries.py", line 172, in post
    return await self._post_impl(request, data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/config/config_entries.py", line 179, in _post_impl
    return await super()._post_impl(request, data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/data_entry_flow.py", line 84, in _post_impl
    result = await self._flow_mgr.async_init(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1287, in async_init
    flow, result = await self._async_init(flow_id, handler, context, data)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1312, in _async_init
    flow = await self.async_create_flow(handler, context=context, data=data)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 1499, in async_create_flow
    handler = await _async_get_flow_handler(
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 2959, in _async_get_flow_handler
    await _load_integration(hass, domain, hass_config)
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 2936, in _load_integration
    await async_process_deps_reqs(hass, hass_config, integration)
  File "/usr/src/homeassistant/homeassistant/setup.py", line 575, in async_process_deps_reqs
    await requirements.async_get_integration_with_requirements(
  File "/usr/src/homeassistant/homeassistant/requirements.py", line 54, in async_get_integration_with_requirements
    return await manager.async_get_integration_with_requirements(domain)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/requirements.py", line 170, in async_get_integration_with_requirements
    await future
  File "/usr/src/homeassistant/homeassistant/requirements.py", line 159, in async_get_integration_with_requirements
    await self._async_process_integration(integration, done)
  File "/usr/src/homeassistant/homeassistant/requirements.py", line 182, in _async_process_integration
    await self.async_process_requirements(
  File "/usr/src/homeassistant/homeassistant/requirements.py", line 269, in async_process_requirements
    await self._async_process_requirements(name, missing)
  File "/usr/src/homeassistant/homeassistant/requirements.py", line 307, in _async_process_requirements
    raise RequirementsNotFound(name, list(failures))
homeassistant.requirements.RequirementsNotFound: Requirements for gruenbeck_cloud not found: ['pygruenbeck_cloud==0.1.0'].
```

### Diagnostics dump

cannot provide diagnostics because integration no longer apears in integration panel:
```
Fehler

Der Konfigurationsfluss konnte nicht geladen werden: 500 Internal Server Error Server got itself in trouble

```

**Comment by p0l0** [comment]
This is duplicated with issue #129, it is fixed in the new version 1.0.0 which I try to release this week.

**Comment by jubie25** [comment]
Can confirm this: after updating to beta 1.000 the integration could be installed again. The device could be reactivated and I got things back running. 
Thanks a lot!
